src/main/java/api/store/diglog/service/FolderService.java

```diff
@@ -0,0 +1,149 @@
+package api.store.diglog.service;
+
+import static api.store.diglog.common.exception.ErrorCode.*;
+
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.TreeMap;
+import java.util.UUID;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import api.store.diglog.common.exception.CustomException;
+import api.store.diglog.model.dto.folder.FolderCreateRequest;
+import api.store.diglog.model.dto.folder.FolderResponse;
+import api.store.diglog.model.entity.Folder;
+import api.store.diglog.model.entity.Member;
+import api.store.diglog.repository.FolderRepository;
+import lombok.RequiredArgsConstructor;
+
+@Service
+@RequiredArgsConstructor
+@Transactional(readOnly = true)
+public class FolderService {
+
+	private static final int MAX_FOLDER_SIZE = 100;
+
+	private final FolderRepository folderRepository;
+	private final MemberService memberService;
+
+	@Transactional
+	public List<FolderResponse> createAndUpdateFolders(List<FolderCreateRequest> folderCreateRequests) {
+
+		validateDuplicationParentFolderAndTitle(folderCreateRequests);
+		validateFolderSize(folderCreateRequests);
+		validateDuplicationOrderIndex(folderCreateRequests);
+
+		Member member = memberService.getCurrentMember();
+		Map<Integer, Folder> allFolders = new TreeMap<>();
+
+		int maxDepth = calculateMaxDepth(folderCreateRequests);
+		IntStream.range(0, maxDepth + 1)
+			.forEach(depth -> {
+				Map<Integer, Folder> foldersAtCurrentDepth = createAndUpdateFoldersByDepth(
+					folderCreateRequests,
+					member,
+					depth,
+					allFolders
+				);
+
+				allFolders.putAll(foldersAtCurrentDepth);
+			});
+
+		List<Folder> savedFolders = folderRepository.saveAll(allFolders.values());
+		return savedFolders.stream()
+			.map(folder -> FolderResponse.builder()
+				.folder(folder)
+				.build())
+			.toList();
+	}
+
+	private void validateDuplicationParentFolderAndTitle(List<FolderCreateRequest> folderCreateRequests) {
+
+		Set<String> uniqueFolders = new HashSet<>();
+
+		folderCreateRequests.forEach(request -> {
+			if (!uniqueFolders.add(request.getParentOrderIndex() + request.getTitle())) {
```

Comment:
```suggestion
if (!uniqueFolders.add(request.getParentOrderIndex() + "-" + request.getTitle())) {
```
매우 희박한 경우지만, 1 + "11"과 11 + "1"과 같이 의도치 않은 Exception이 발생할 수 있을 것 같아서 "-"과 같은 문자를 추가해주면 좋을 것 같습니다!
덧붙여, Set을 사용해서 중복 검사를 하는 아이디어는 정말 좋은 것 같습니다 😊

Reply:
생각하지 못했던 케이스였습니다 😢 수정 후 반영하겠습니다!!